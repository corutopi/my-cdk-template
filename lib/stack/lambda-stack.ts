import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { IamRole } from '../resource/lambda/iam-role';
import { LambdaFunction } from '../resource/lambda/lambda-function';

/**
 * LambdakStack を作成するクラス.
 */
export class LambdakStack extends cdk.Stack {
  public readonly iamRole: IamRole;
  public readonly lambdaFunction: LambdaFunction;

  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    this.iamRole = new IamRole({ scope: this });
    this.lambdaFunction = new LambdaFunction({ scope: this }, { iamRole: this.iamRole });
  }
}
