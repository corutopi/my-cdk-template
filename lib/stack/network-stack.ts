import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { BaseStack, BaseStackProps } from './abstruct/base-stack';
import { Vpc } from '../resource/network/vpc';
import { Subnet } from '../resource/network/subnet';
import { InternetGateway } from '../resource/network/internet-gateway';
import { RouteTable } from '../resource/network/route-table';

/**
 * NetworkStack を作成するクラス.
 */
export class NetworkStack extends BaseStack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;
  public readonly internetGateway: InternetGateway;
  public readonly routeTable: RouteTable;

  constructor(parentProps: BaseStackProps) {
    super(parentProps);

    this.vpc = new Vpc({ scope: this });
    this.subnet = new Subnet({ scope: this }, { vpc: this.vpc });
    this.internetGateway = new InternetGateway({ scope: this }, { vpc: this.vpc });
    this.routeTable = new RouteTable(
      { scope: this },
      { vpc: this.vpc, subent: this.subnet, internetGateway: this.internetGateway }
    );
  }
}
