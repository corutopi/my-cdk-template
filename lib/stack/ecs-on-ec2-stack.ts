import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { IamRole } from '../resource/ecs-on-ec2/iam-role';
import { SecurityGroup } from '../resource/ecs-on-ec2/security-group';
import { Instance } from '../resource/ecs-on-ec2/ec2-instance';
import { EcsCluster } from '../resource/ecs-on-ec2/ecs-cluster';
import { NetworkStack } from '../stack/network-stack';

/**
 * NetworkStack を作成するクラス.
 */
export class EcsOnEc2Stack extends cdk.Stack {
  public readonly role: IamRole;
  public readonly sg: SecurityGroup;
  public readonly ins: Instance;
  public readonly cluster: EcsCluster;

  constructor(scope: Construct, id: string, networkStack: NetworkStack, props?: cdk.StackProps) {
    super(scope, id, props);
    this.role = new IamRole({ scope: this });
    this.sg = new SecurityGroup({ scope: this }, { vpc: networkStack.vpc });
    this.ins = new Instance(
      { scope: this },
      { role: this.role, sg: this.sg, subnet: networkStack.subnet }
    );
    this.cluster = new EcsCluster({ scope: this });
  }
}
