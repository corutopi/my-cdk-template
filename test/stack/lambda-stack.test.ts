import * as cdk from 'aws-cdk-lib';
import { Template, Match } from 'aws-cdk-lib/assertions';

import { LambdakStack } from '../../lib/stack/lambda-stack';
import * as tc from '../test-constant';

/**
 * テストで使用するテンプレートを生成する.
 *
 * @returns 生成した Template インスタンス
 */
function createTestTemplate(): Template {
  const app = new cdk.App({ context: tc.CONTEXT });
  const stack = new LambdakStack({ id: 'lambda', scope: app });
  const template = Template.fromStack(stack);
  return template;
}

test('Snapshot', () => {
  const template = createTestTemplate();
  const json = template.toJSON();

  expect(json).toMatchSnapshot('LambdaStackSnapshot');
});

test('IamRole', () => {
  const template = createTestTemplate();

  template.resourceCountIs('AWS::IAM::Role', 1);
  template.hasResourceProperties('AWS::IAM::Role', {
    AssumeRolePolicyDocument: {
      Statement: [
        {
          Action: 'sts:AssumeRole',
          Effect: 'Allow',
          Principal: {
            Service: 'lambda.amazonaws.com',
          },
        },
      ],
      Version: '2012-10-17',
    },
    ManagedPolicyArns: ['arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'],
    RoleName: 'cdktest-dev-test-function-role',
  });
});

test('LambdaFunction', () => {
  const template = createTestTemplate();

  template.resourceCountIs('AWS::Lambda::Function', 1);
  template.hasResourceProperties('AWS::Lambda::Function', {
    Code: {
      S3Bucket: 'resourceBucket',
      S3Key: 'test-function.zip',
    },
    Role: {
      'Fn::GetAtt': ['IamRoleTestFunction', 'Arn'],
    },
    FunctionName: 'cdktest-dev-test-func',
    Handler: 'lambda_function.lambda_handler',
    Runtime: 'python3.8',
  });
  template.resourceCountIs('AWS::Lambda::Alias', 1);
  template.hasResourceProperties('AWS::Lambda::Alias', {
    FunctionName: 'cdktest-dev-test-func',
    FunctionVersion: '$LATEST',
    Name: 'InService',
  });
  template.resourceCountIs('AWS::Lambda::Version', 1);
  template.hasResource('AWS::Lambda::Version', {
    Properties: {
      FunctionName: 'cdktest-dev-test-func',
      Description: Match.anyValue(),
    },
    DependsOn: ['LambdaFunctionTest'],
    UpdateReplacePolicy: 'Retain',
    DeletionPolicy: 'Retain',
  });
  const f = template.findResources('AWS::Lambda::Version', {
    Properties: {
      FunctionName: 'cdktest-dev-test-func',
      Description: Match.anyValue(),
    },
    DependsOn: ['LambdaFunctionTest'],
    UpdateReplacePolicy: 'Retain',
    DeletionPolicy: 'Retain',
  });
});

test('Logs', () => {
  const template = createTestTemplate();

  template.resourceCountIs('AWS::Logs::LogGroup', 1);
  template.hasResourceProperties('AWS::Logs::LogGroup', {
    LogGroupName: '/aws/lambda/cdktest-dev-test-func',
    RetentionInDays: 90,
  });
});
