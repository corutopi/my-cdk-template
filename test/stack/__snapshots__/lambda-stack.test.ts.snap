// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: LambdaStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "CloudWatchLogsMain": Object {
      "Properties": Object {
        "LogGroupName": "/aws/lambda/cdktest-dev-test-func",
        "RetentionInDays": 90,
      },
      "Type": "AWS::Logs::LogGroup",
    },
    "IamRoleTestFunction": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
        ],
        "RoleName": "cdktest-dev-test-function-role",
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaFunctionTest": Object {
      "Properties": Object {
        "Code": Object {
          "S3Bucket": "resourceBucket",
          "S3Key": "test-function.zip",
        },
        "FunctionName": "cdktest-dev-test-func",
        "Handler": "lambda_function.lambda_handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "IamRoleTestFunction",
            "Arn",
          ],
        },
        "Runtime": "python3.8",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaFunctionTestAliasInService": Object {
      "DependsOn": Array [
        "LambdaFunctionTest",
      ],
      "Properties": Object {
        "FunctionName": "cdktest-dev-test-func",
        "FunctionVersion": "$LATEST",
        "Name": "InService",
      },
      "Type": "AWS::Lambda::Alias",
    },
    "LambdaFunctionTestVersion3f9eebb283b6a298918ccaa48964c2ac": Object {
      "DeletionPolicy": "Retain",
      "DependsOn": Array [
        "LambdaFunctionTest",
      ],
      "Properties": Object {
        "Description": "When the version changes, the comments must also change.",
        "FunctionName": "cdktest-dev-test-func",
      },
      "Type": "AWS::Lambda::Version",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
